content/postgresql/concepts/window-functions/window-functions.md

```diff
@@ -0,0 +1,101 @@
+
+|        Function        |      Syntax      |      Description     |
+|          ---           |       ---        |         ----         |
+|Rank() |`SELECT column_name1, column_name2, column_nameN, rank() OVER (PARTITION BY column_name ORDER BY column_name DESC) FROM table_name;`|Returns the ranking of the current row.|
+|Row_number()|`SELECT column_name1, column_name2, column_nameN, row_number () OVER (PARTITION BY column_name ORDER BY column_name DESC) FROM table_name;`|Returns the number of the current row within its partion, counting from 1.|
+| Dense rank()|`SELECT column_name1, column_name2, column_nameN, dense_rank () OVER (PARTITION BY column_name ORDER BY column_name DESC) FROM table_name;`| Returns the rank of the current row, this function effectively counts groups.|
+| Ntile()| `SELECT column_name1, column_name2, column_nameN, ntile (Argument) OVER (PARTITION BY column_name ORDER BY column_name DESC) FROM table_name;`|Returns an integer ranging from 1 to the argument value.|
+|Cume dist()| `SELECT column_name1, column_name2, column_nameN, cume_dist () OVER (PARTITION BY column_name ORDER BY column_name ASC) FROM table_name;`|Returns the cumulative distribution.|  
+| Percent rank()| `SELECT column_name1, column_name2, column_nameN, percent_rank () OVER (PARTITION BY column_name ORDER BY column_name ASC) FROM table_name;`|Returns the relative rank of the current row, that is the total partition rows - 1.|
+| First Value()|`SELECT column_name1, column_name2, column_nameN, first_value (value) OVER (PARTITION BY column_name ORDER BY column_name ASC) FROM table_name;`|Returns the -value- of the first row in a window frame.|
+|Last value()|`SELECT column_name1, column_name2, column_nameN, last_value (value) OVER (PARTITION BY column_name ORDER BY column_name ASC) FROM table_name;`|Returns the -value- of the last row in a window frame.|
+
+## Example
+
+To understand the `Rank()` and `Row_number()` functions, let's consider the following _staff_ table as an example to demonstrate how these window functions work:
+
+| id | employee | adrress | contact | salary | 
+|--- | -----    | -----   | -----   | ----   |
+| 01 | John     | Tokyo   | +81     | 5500   |
+| 02 | Kate     | Beijing | +186    | 5000   |
+| 03 | Tad      | Mumbai  | +91     | 2500   |
+| 04 | Martha   | Tokyo   | +727    | 3500   |
+| 05 | Will     | paris   | +33     | 4000   |
+| 06 | Ruth     | Dubai   | +971    | 2500   |
+| 07 | Kaka     | Mumbai  | +213    | 4500   |
+| 08 | Tolf     | Tokyo   | +44     | 2000   |
+
+From the above _staff_ table, to rank the employees using the `Rank()` function, we select the id, salary, and address columns from the _staff_ table, and then use the `RANK()` window function to assign a rank to each row based on the _salary_ column within each partition defined by the _address_ column.
+
+#Syntax:
+
+`SELECT id, salary, address, Rank() OVER (PARTITION BY address ORDER BY salary DESC) FROM staff;`
```

Comment:
```suggestion

SELECT id, salary, address, Rank() OVER (PARTITION BY address ORDER BY salary DESC) FROM staff;
```
This is not a syntax as we are using it to print something. Write the code in the code block (```sql). Check the [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md) for the reference.